WBooks/Library/MainMenuController.swift

```diff
@@ -39,15 +39,11 @@ class MainMenuController: UIViewController {
         configureLibraryNavBar()
     }
     
-    func onSuccess(books: [Book]) {
-        _viewModel.onSuccess(books: books)
-    }
-    
     private func configureTableView() {
-        _viewModel.books.producer.startWithValues { [weak self] _ in
+        _viewModel.state.producer.startWithValues { [weak self] _ in
             self?._view.tableView.reloadData()
```

Comment:
You are reloading the tableView each time `state` changes. Shouldn't be only when the array is not empty?

Reply:
Indeed. I'll change it!